### ISSUE TITLE
sops metadata not found

### ISSUE THREAD

**Issue by domingusj** [futuresimple/helm-secrets#18]
I keep getting `Error loading file metadata: sops metadata not found` when trying to interact with any secret files using `helm-wrapper` (see output below). I wonder if it is an issue with sops v3, or using GCP KMS, or if I am missing any config. Any help would be appreciated!

```
>   sops -v                                                                                          ---
sops 3.0.0 (latest)
------------------------------------------------------------
>   cat .sops.yaml                                                                                   ---
creation_rules:
    - gcp-kms: 'projects/my-gcp-project/locations/global/keyRings/sops/cryptoKeys/my-app'
------------------------------------------------------------
>   cat k8s/my-app/secrets.yaml                                                                                 ---
SECRET_1: ENC[AES256_GCM,data:<data>,iv:<iv>=,tag:<tag>==,type:str]
SECRET_2: ENC[AES256_GCM,data:<data>,iv:<iv>=,tag:<tag>==,type:str]
SECRET_3: ENC[AES256_GCM,data:<data>,iv:<iv>=,tag:<tag>==,type:str]
sops:
    kms: []
    gcp_kms:
    -   resource_id: projects/cz-kms/locations/global/keyRings/sops/cryptoKeys/my-app
        created_at: '2017-10-12T02:50:29Z'
        enc: <enc_data>
    lastmodified: '2017-10-12T02:50:48Z'
    mac: <enc_data>
    pgp: []
    unencrypted_suffix: _unencrypted
    version: 3.0.0
------------------------------------------------------------
>   sops -d k8s/my-app/secrets.yaml                                                            ---
[GCPKMS]	 WARN[0000] Decryption succeeded                          resourceID=projects/cz-kms/locations/global/keyRings/sops/cryptoKeys/my-app
[SOPS]	 INFO[0000] Data key recovered successfully
SECRET_1: foo
SECRET_2: bar
SECRET_3: baz
------------------------------------------------------------
>   helm-wrapper secrets view k8s/my-app/secrets.yaml                                          ---

Error loading file metadata: sops metadata not found
Error: plugin "secrets" exited with error

------------------------------------------------------------
```

**Comment by szibis** [comment]
I can't reproduce.

```
helm-wrapper secrets view example/helm_vars/secrets.yaml

[PGP]	 INFO[0000] Decryption succeeded                          fingerprint=4434EA5D05F10F59D0DF7399AF1D073646ED4927
[SOPS]	 INFO[0000] Data key recovered successfully
global_secret: global_bar
```

All files from test vars decrypting and tests are passing.

```
+++ Testing ./example/helm_vars/projectX/production/us-east-1/java-app/secrets.yaml
+++ Encrypt and Test
[OK] File properly encrypted
+++ Test if 'Already Encrypted' feature works
[OK] Already Encrypted
+++ View encrypted Test
[PGP]	 INFO[0000] Decryption succeeded                          fingerprint=4434EA5D05F10F59D0DF7399AF1D073646ED4927
[SOPS]	 INFO[0000] Data key recovered successfully
[OK] File decrypted and viewable
+++ Decrypt
[PGP]	 INFO[0000] Decryption succeeded                          fingerprint=4434EA5D05F10F59D0DF7399AF1D073646ED4927
[SOPS]	 INFO[0000] Data key recovered successfully
[OK] File decrypted
+++ Cleanup Test
[OK] Cleanup specified directory
[OK] Cleanup specified .dec file
[OK] Cleanup specified encrypted secret file
+++ Once again Encrypt and Test
[PGP]	 INFO[0001] Encryption succeeded                          fingerprint=4434EA5D05F10F59D0DF7399AF1D073646ED4927
```

**Comment by szibis** [comment]
```
cat  ./example/helm_vars/projectX/production/us-east-1/java-app/secrets.yaml
secret_production_projectx: ENC[AES256_GCM,data:q3OzzTF7IYAHIiDiz5U=,iv:hOrad3FgXdJEWvSFwcLApltEPU07fsNSzB6rHLp74oc=,tag:LFcngS1VSXZjxMMS4ue+sQ==,type:str]
sops:
    kms: []
    gcp_kms: []
    lastmodified: '2017-11-02T22:01:01Z'
    mac: ENC[AES256_GCM,data:MGvrYH/kXZEZuOS5hjTRWzhSaNbiaURDygwJcGJ9qW0Duv0WjoBs9Cm02dw+HNq+YhOG0rrqtA6r1CvTrI59tB2Xwq6s72brIZPsHkYp5bWD9+p7F3rVkxyVhF70/o+Q6cPxxkuNUpg0KdaujfsHcITYQkS7R4Vy+7NlSFP8UUk=,iv:t3y+9G0uHipF7VpuvNK2lzYGrfU5vRkd2/5ZRqIYz50=,tag:FZ9RTsyZTJTxkMDqsVQyLA==,type:str]
    pgp:
    -   created_at: '2017-11-02T22:01:00Z'
        enc: |-
            -----BEGIN PGP MESSAGE-----

            wcFMAxYpv4YXKfBAARAAhDNN4UXwq0FvWUX+S4Q15IeEb24efBpF7vP2zUz9UBAM
            SiHmToyB8FcOMeHIQACZxM8BGzuc0IDoZUfnsDrNMyNCxbiddXMMB+L8Z7T3QzfL
            IzDj20kVPSrsgzYmfQ1eZ/WIRonXSSQ/c4ZP/7dtRHZ38qFFwFua8JtEmTPWeSUq
            tIv6ZmKQ73Gzmbp06kVB412fMP8ZMIVzP5XOMX1uTUPl8hvOY/gqGI0qMpyZ4ZBk
            woFvby4BjkdHlCJZjFbWHwpIwYvSyOF6tm+Ew9Dp9OK7ogpDAdHhdPTHl+AGKPSB
            ov76E/pnEkJf10U7gEpPJWuNYGt5kQWqyCy4+hv8FlXCjXI+YV8ucI+vPvOBw6Ok
            OSsKOIWJNnS4ZxJbzZnHWT3b5HYIDVNLyPupwSNemNsEjSzxSjLIyYrbI2WfU+N+
            dhclW9nD1PWuhLAtvYAs1tjNiH7uG7gZqjieEDF0+muJ0PGdwxEdlFQ2ih71bfsZ
            ZyzUPuV8VVw0hIbUxev17zlJYV+ZTmVsPziOJQOB+/9zB8YjCybON6wXMUPVAv6f
            sNjbyCj4ub2mioCqdheWZphg0CAFMYnzK6fUYLPF+uRHPCHZrBhG5vVTbwPJrBkh
            HeFc2jfnNbZSFOWJyQ/Osll3oPAD3lEZZD6QzfiSR28ieWq/+qa2+BFATjDINejS
            4AHkE/bm/FCbM2YfJmGClGJ01eGR0OAc4EDhs4vgceINK5lT4C/lgCwI5jiFH+DV
            jYeiufTT0OE/88kwCTnEah3GzarCWILgx+S3OYj7eaT+lpRG9GDiGlRg4pC+oMjh
            g3kA
            =WURo
            -----END PGP MESSAGE-----
        fp: 4434EA5D05F10F59D0DF7399AF1D073646ED4927
    unencrypted_suffix: _unencrypted
    version: 3.0.0
```

**Comment by domingusj** [comment]
Thanks for the response. In your example did it actually use `GCPKMS` or `PGP`? (`[PGP]	 INFO[0000] Decryption succeeded `) I was able to get it working with `PGP` but with `GCPKMS` I still get ```Error loading file metadata: sops metadata not found
Error: plugin "secrets" exited with error```

**Comment by mumoshu** [comment]
@domingusj Hi! Just curious but are you sure you have a valid credentials.json(?) to authenticate against GCP API?
The exact error occurred for me in the past - I was using AWS KMS, had a valid KMS key ARN in the `SOPS_KMS_ARN` env var, but missed a valid session against AWS API.

**Comment by domingusj** [comment]
Well, just the plain `sops` command works with GCP KMS. For now I just switched to using palin `sops` and the same folder + file structure outlined with `helm-secrets` and I have a manual decrypt step. I'll try to dig into it more this weekend.

**Comment by snebel29** [comment]
Hi,
The TL; DR is rename `gcp-kms` to `gcp_kms` in your `.sops.yaml` file

I bet you encrypted that file using sops with gcp kms but not using the `.sops.yaml` file described above, then this was failing when trying to decrypt because of the wrong key being used, I was also using that wrong key name (I'm not sure why) and producing encrypted files without any metadata into the sop metadata section for that backend, then I realize of the silly mistake I was doing.

You can confirm the key name in both any encrypted file (even if is using other encryption backend, you will see it as empty however) and into [this line within sops code itself](https://github.com/mozilla/sops/blob/d7c63ec1e28612ef168412baa6596f3b9cbb118a/stores/stores.go#L40
)

Also In general GCP kms is under documented in both this project and sops, I might create some PR to help documenting this now that I'm planing on start using this.

Bye

**Comment by domingusj** [comment]
Thank you! Very much appreciated!

**Comment by nuthankumar** [comment]
Hi guys... I do not see helm-wrapper as part of installation of helm secret plugin install. 
help is much appreciated

**Comment by daanlevi** [comment]
It is no longer needed since it is part of helm command now.
helm secrets edit secrets.yaml should be enough.
